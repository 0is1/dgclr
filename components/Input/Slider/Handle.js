// @flow
import React, { PureComponent } from 'react';
import Styles from './Handle.styles';
import SliderRailStyles from './SliderRail.styles';

const { TooltipContainer, Tooltip, TooltipText } = SliderRailStyles;

const { HandleDiv, Slider } = Styles;

type Props = {
  domain: Array<number>,
  isActive: boolean,
  handle: {
    id: string,
    value: number,
    percent: number,
  },
  getHandleProps: Function,
  disabled?: boolean,
  format: Function,
};
type State = { mouseOver: boolean };

class Handle extends PureComponent<Props, State> {
  static defaultProps = { disabled: false };

  state = {
    mouseOver: false,
  };

  onMouseEnter = () => {
    this.setState({ mouseOver: true });
  };

  onMouseLeave = () => {
    this.setState({ mouseOver: false });
  };

  render() {
    const {
      domain: [min, max],
      handle: { id, value, percent },
      isActive,
      disabled,
      getHandleProps,
      format,
    } = this.props;
    const { mouseOver } = this.state;
    return (
      <>
        {(mouseOver || isActive) && !disabled ? (
          <TooltipContainer percent={percent}>
            <Tooltip>
              <TooltipText>{format(value)}</TooltipText>
            </Tooltip>
          </TooltipContainer>
        ) : null}
        <HandleDiv
          percent={percent}
          {...getHandleProps(id, {
            onMouseEnter: this.onMouseEnter,
            onMouseLeave: this.onMouseLeave,
          })}
        />
        <Slider
          role="slider"
          aria-valuemin={min}
          aria-valuemax={max}
          aria-valuenow={value}
          percent={percent}
          disabled={disabled}
        />
      </>
    );
  }
}

Handle.defaultProps = {
  disabled: false,
};

export default Handle;
