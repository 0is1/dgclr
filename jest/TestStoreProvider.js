// @flow
/* eslint-env jest */
import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import configureStore from 'redux-mock-store';

import type { Node } from 'react';

export const mockStore = configureStore();

type Props = {
  children: Node,
  storeData: {},
};

function TestStoreProvider({ children, storeData }: Props) {
  const store = mockStore(storeData);
  return <Provider store={store}>{children}</Provider>;
}

export default TestStoreProvider;
