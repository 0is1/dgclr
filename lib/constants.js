export const ADVANCED_COURSE_INFO = 'courseInfo';
export const ADVANCED_NEARBY = 'nearby';

export const RATING_OPTIONS = [
  { value: 'AAA1', label: 'AAA1' },
  { value: 'AA1', label: 'AA1' },
  { value: 'A1', label: 'A1' },
  { value: 'AA2', label: 'AA2' },
  { value: 'AA3', label: 'AA3' },
  { value: 'A2', label: 'A2' },
  { value: 'A3', label: 'A3' },
  { value: 'BB1', label: 'BB1' },
  { value: 'BB2', label: 'BB2' },
  { value: 'B1', label: 'B1' },
  { value: 'B2', label: 'B2' },
  { value: 'B3', label: 'B3' },
  { value: 'C1', label: 'C1' },
  { value: 'C2', label: 'C2' },
  { value: 'C3', label: 'C3' },
  { value: 'D1', label: 'D1' },
  { value: 'D2', label: 'D2' },
  { value: 'D3', label: 'D3' },
];

export const RATINGS_SORT_ORDER = [
  'AAA1',
  'AA1',
  'A1',
  'AA2',
  'AA3',
  'A2',
  'A3',
  'BB1',
  'BB2',
  'B1',
  'B2',
  'B3',
  'C1',
  'C2',
  'C3',
  'D1',
  'D2',
  'D3',
];

export const MAP_DEFAULT_ZOOM = 9;

export const SERVER_URL = 'https://dgclr-server.herokuapp.com/graphql';
