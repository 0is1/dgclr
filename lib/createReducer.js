/**
 * Avoid huge switch-case based reducers
 *
 * This
 *
 *     function reducer(state, action) {
 *         switch(action.type) {
 *         case OPEN:
 *             return {...state, open: true};
 *         case CLOSE:
 *             return {...state, open: false};
 *          default:
 *            return state;
 *         }
 *     }
 *
 * can be written as
 *
 *     const reducer = createReducer({
 *         [OPEN]: (state, action) => ({...state, open: true}),
 *         [CLOSE]: (state, action) => ({...state, open: false}),
 *     });
 *
 *
 * Signature is
 *
 * @param {Object} mapping - Mapping of actions to reducers
 *
 * or
 *
 * @param {String} action - Single action
 * @param {Function} reducer - Reducer for the single action
 *
 *
 * @return {Function} Reducer function
 */
export function createReducer(arg1, arg2, defaultState = null) {
  let mapping = null;

  if (typeof arg1 === 'string') {
    mapping = { [arg1]: arg2 };
  } else {
    mapping = arg1;
  }

  return (state = defaultState, action) => {
    const reducer = mapping[action.type];
    if (typeof reducer === 'function') {
      const res = reducer(state, action);
      if (typeof res === 'function') {
        return res(state);
      }
      return res;
    }

    return state;
  };
}

export default createReducer;
