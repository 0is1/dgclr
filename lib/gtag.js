import { v4 as uuidv4 } from 'uuid';

const { NODE_ENV } = process.env;

export const GA_TRACKING_ID = 'UA-124877413-1';

// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
export const pageview = (url) => {
  // Track only in production
  if (NODE_ENV && NODE_ENV === 'production') {
    const uuid = uuidv4();
    window.gtag('config', GA_TRACKING_ID, {
      page_location: url,
      custom_map: {
        dimension1: 'browser_id',
        dimension2: 'utc_millisecs',
      },
    });
    window.gtag('event', 'browser_id_dimension', { browser_id: uuid });
    window.gtag('event', 'utc_millisecs_dimension', { utc_millisecs: new Date().getTime() });
  }
};

// https://developers.google.com/analytics/devguides/collection/gtagjs/events
export const event = ({
  action, category, label, value,
}) => {
  // Track only in production
  if (NODE_ENV && NODE_ENV === 'production') {
    window.gtag('event', action, {
      event_category: category,
      event_label: label,
      value,
    });
  }
};
