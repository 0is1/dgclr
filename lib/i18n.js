const NextI18Next = require('next-i18next').default;

const NextI18NextInstance = new NextI18Next({
  defaultLanguage: 'fi',
  otherLanguages: ['en'],
});

module.exports = NextI18NextInstance;
