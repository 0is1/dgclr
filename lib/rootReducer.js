import { combineReducers } from 'redux';

import search, { initialState as searchInitialState } from 'components/SearchContainer/reducers';

export const DEFAULT_STATE = {
  search: searchInitialState,
};

export default combineReducers({
  search,
});
