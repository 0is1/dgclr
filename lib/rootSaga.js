import { all } from 'redux-saga/effects';

import search from 'components/SearchContainer/sagas';

function* rootSaga() {
  yield all([search]);
}

export default rootSaga;
