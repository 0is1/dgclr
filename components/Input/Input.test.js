/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { Slider } from 'react-compound-slider';
import TestStoreProvider from 'jest/TestStoreProvider';
import { DEFAULT_STATE } from 'lib/rootReducer';
import InputComponent from './Input';
import Handle from './Slider/Handle';
import Styles from './Input.styles';

const { Input } = Styles;

const mockProps = (props = {}) => ({
  focusOnMount: false,
  options: { type: 'text' },
  placeholder: '',
  value: '',
  onChange: jest.fn(),
  ...props,
});

const setup = (props = {}) => {
  const storeData = {
    ...DEFAULT_STATE,
  };
  const components = (
    <TestStoreProvider storeData={storeData}>
      <InputComponent {...props} />
    </TestStoreProvider>
  );
  const enzymeWrapper = mount(components);
  const component = renderer.create(components);
  return { component, enzymeWrapper };
};

describe('Input and Slider component', () => {
  describe('Input', () => {
    it('Match snapshot with default props', () => {
      const props = mockProps();
      const subject = shallow(<InputComponent {...props} />);
      expect(subject).toMatchSnapshot();
    });
    it('Renders correctly', () => {
      const props = mockProps({
        value: 'espoo',
        placeholder: 'Kaupungin tai radan nimi',
      });
      const subject = shallow(<InputComponent {...props} />);
      expect(subject.find(Input).length).toEqual(1);
      expect(subject.find(Input).prop('placeholder')).toEqual('Kaupungin tai radan nimi');
      expect(subject.find(Input).prop('value')).toEqual('espoo');
    });
    it('Does not autofocus by default', () => {
      const props = mockProps();
      const subject = mount(<InputComponent {...props} />);
      const { ref } = subject.instance();
      jest.spyOn(ref.current, 'focus');
      subject.instance().componentDidMount();
      expect(ref.current.focus).toHaveBeenCalledTimes(0);
    });
    it('Autofocus correctly if prop is set', () => {
      const props = mockProps({ focusOnMount: true });
      const subject = mount(<InputComponent {...props} />);
      const { ref } = subject.instance();
      jest.spyOn(ref.current, 'focus');
      subject.instance().componentDidMount();
      expect(ref.current.focus).toHaveBeenCalledTimes(1);
    });
  });
  describe('Slider', () => {
    it('Renders correctly with 1 Handle', () => {
      const options = {
        step: 5,
        type: 'slider',
        initialValues: [0],
        filterName: 'slider-filter',
        domain: [0, 100],
      };
      const props = mockProps({
        options,
      });
      const { component, enzymeWrapper } = setup(props);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      const SliderComponent = enzymeWrapper.find(Slider);
      const { step, domain, values } = SliderComponent.props();
      expect(step).toEqual(5);
      expect(domain).toEqual([0, 100]);
      expect(values).toEqual([0]);
      expect(SliderComponent.length).toEqual(1);
      expect(enzymeWrapper.find(Handle).length).toEqual(1);
    });
    it('Renders correctly with 2 Handles', () => {
      const options = {
        step: 10,
        type: 'slider',
        initialValues: [10, 100],
        filterName: 'slider-filter',
        domain: [0, 200],
      };
      const props = mockProps({
        options,
      });
      const { component, enzymeWrapper } = setup(props);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      const SliderComponent = enzymeWrapper.find(Slider);
      const { step, domain, values } = SliderComponent.props();
      expect(step).toEqual(10);
      expect(domain).toEqual([0, 200]);
      expect(values).toEqual([10, 100]);
      expect(SliderComponent.length).toEqual(1);
      expect(enzymeWrapper.find(Handle).length).toEqual(2);
    });
  });
});
