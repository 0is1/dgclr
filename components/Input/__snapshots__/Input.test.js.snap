// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Match snapshot with default props 1`] = `
<input
  className="sc-eNQAEJ Inputstyles__Input-tjx6pn-0 hRDFua"
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
